### PR TITLE
feat: onDecorateDefinition plugin option

### LINKED ISSUE #99: Recursive object properties
Well, it's me again :smile:  I currently use this plugin heavily to implement the use case already mentioned in #93. I am currently generating mock plugins to perform isolation tests. It would be very helpful if I could not only get the `object` type for decorators but also the properties of this object (already mentioned in #92). Something like this:

```js
app.decorate('objectDecorator', {
  str: 'hello',
  recursiveObj: {}
})
```
would result in
```js
{
  name: 'objectDecorator',
  type: 'object',
  recursive: [
    { name: 'str', type: 'string' },
    { name: 'recursiveObj', type: 'object', recursive: ... }
  ]
}
```

The user of this plugin could control the depth of the recursive object properties by a plugin option like `maxDecoratorDepth` or something similar. If this is outside the scope of this plugin this is fine and I can implement it outside this plugin, but it would be nicer and usable for others within this plugin 😄  If you are open for a PR, I am happy to provide a first implementation :)

Thanks for all your work!

### PR DESCRIPTION
**Pull request by matthyk**
Closes #99

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -318,6 +332,31 @@ override the `source` property.
 ```
 In this example, the `url` property is overridden and the `url` length is returned instead of the `url`.
 
+### onDecorateDefinition
+
+Similar to `onRouteDefinition`, this option allows you to control which information about decorators is included in the overview.
+The passed function is called for `instance`, `request` and `reply` decorators but the decorator type is passed to the function as parameter.
+For the decorator ```app.decorateReply('db', { query: () => {} })```js the `onDecorateDefinition` function will be called like
+`onDecorateDefinition('decorateReply', 'db', { query: () => {} })`. The default properties `name` and `type` can also be overwritten here.
```
Comment (Eomm) [inline README.md:340]: This could be an MD table for clarity mentioning 'decorate' | 'decorateRequest' | 'decorateReply'

File: README.md [hunk 2]
```diff
@@ -176,11 +176,25 @@ app.register(require('fastify-overview'), {
...
+           recursive: Object.entries(value).map(([key, val]) => {
+              return {
+                 name: key,
+                 type: Array.isArray(val) ? 'array' : typeof val
+              }
+           })
+        }
+     } else {
+        return {}
+     }
```
Comment (Eomm) [inline README.md:197]: I would not repeat the same example as below

Here I would keep it simple:

```suggestion
     if (value && typeof value === 'object' && !Array.isArray(value)) {
        return {
           staticData: true
        }
     }
     return { utilityFunction: true }
```

File: README.md [hunk 3]
```diff
@@ -176,11 +176,25 @@ app.register(require('fastify-overview'), {
...
     url: '/customUrl', // default: '/json-overview'
-  },
-   onRouteDefinition: (opts) => {
+  }, 
+  onRouteDefinition: (opts) => {
     return {
       schema: opts.schema
     }
+  }, 
+  onDecorateDefinition: (type, name, value) => {
```
Comment (Eomm) [inline README.md:185]:
```suggestion
  onDecorateDefinition: (decoratorType, decoratorName, decoratorValue) => {
```

**Review by Eomm — COMMENTED** [review]
LGTM

Minor suggestions to improve the docs 👍🏼

File: README.md (hunk 1 above)
Comment (matthyk) [reply README.md]: Done!

**Review by Eomm — APPROVED** [review]

**Pull request merged**